CliFx/Domain/CommandArgumentSchema.cs

```diff
@@ -176,4 +159,71 @@ internal partial class CommandArgumentSchema
                 BindingFlags.Public | BindingFlags.Static,
                 null, new[] {typeof(string), typeof(IFormatProvider)}, null);
     }
+
+    // Default and valid value handling.
+    internal partial class CommandArgumentSchema
+    {
+        public IReadOnlyList<string> GetValidValues()
+        {
+            var result = new List<string>();
+
+            // Some arguments may have this as null due to a hack that enables built-in options
+            if (Property == null)
+                return result;
+
+            var underlyingPropertyType =
+                Property.PropertyType.GetNullableUnderlyingType() ?? Property.PropertyType;
+
+            // Enum
+            if (underlyingPropertyType.IsEnum)
+                result.AddRange(Enum.GetNames(underlyingPropertyType));
+
+            return result;
+        }
+
+        public string? GetDefaultValue(object? instance)
+        {
+            if (Property is null || instance is null)
+            {
+                return null;
+            }
+
+            var instanceType = instance.GetType();
+            var commandType = Property?.DeclaringType;
+
+            if (instanceType != commandType)
+            {
+                throw new ArgumentException($"Argument {nameof(instance)} of type {instanceType} " +
+                    $"must be the same as this command's type {nameof(commandType)}.");
+            }
```

Comment:
We can drop this check and just assume that the provided type is correct (because it's only used internally). We have tests for this already anyway,

Reply:
Yup that makes sense.